projects/sandbox/datagen/datagen/scripts/timeslide_waveforms.py

```diff
@@ -154,6 +154,19 @@ def main(
     rejected_params = InjectionParameterSet()
     while n_samples > 0:
         params = prior.sample(n_samples)
+
+        # If a Bilby PriorDict has a conversion function, any
+        # extra keys generated by the conversion function will
+        # be added into the sampled parameters, but only if
+        # we're sampling exactly 1 time. This removes those
+        # extra keys
+        if n_samples == 1:
+            extra_keys = [
```

Comment:
overkill, but might be simpler to just go `params = {k: params[k] for k in parameters}`

Might also be worth adding a TODO to ditch this if/when [this change](https://git.ligo.org/lscsoft/bilby/-/merge_requests/1286) makes it into the lib

Reply:
Yeah, no, good point, added both

Reply:
Forgot that `parameters` has some keys that params won't have, so I corrected for that. Also, I _gotta_ stop pushing changes to `datagen` without testing them just because they seem simple.